packages/api/src/models/Validator.js

```diff
@@ -1,12 +1,46 @@
+const BlockHeader = require('./BlockHeader')
+
 module.exports = class Validator {
   proTxHash
+  proposedBlocksAmount
+  lastProposedBlockHeader
 
-  constructor (proTxHash) {
+  constructor (
+    proTxHash,
+    proposedBlocksAmount,
+    lastProposedBlockHeader
+  ) {
     this.proTxHash = proTxHash
+    this.proposedBlocksAmount = Number(proposedBlocksAmount)
```

Comment:
Casting should happen in the fromRaw, I believe. Class constuctor expects arguments of specific data types

Reply:
✅